pkg/controllers/cloud_config_sync_controller.go

```diff
@@ -74,8 +78,18 @@ func (r *CloudConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		return ctrl.Result{}, err
 	}
-	sourceCM, err = r.prepareSourceConfigMap(sourceCM, infra)
+
+	cloudConfigTransformerFn, err := cloud.GetCloudConfigTransformer(infra.Status.PlatformStatus)
 	if err != nil {
+		klog.Errorf("unable to get cloud config transformer function; unsupported platform")
+		if err := r.setDegradedCondition(ctx); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to set conditions for cloud config controller: %v", err)
+		}
+		return ctrl.Result{}, err
+	}
+
+	if sourceCM, err = cloudConfigTransformerFn(sourceCM, infra); err != nil {
```

Comment:
The fact we are reusing the variable here makes me think we don't need to return the configmap at all from the function